### PR TITLE
Blog post widget manifesto

### PR DESCRIPTION
**Pull request by pavpanchekha**
I don't think our bar for blog posts should be that high, but perhaps @chrishtr you can give it the once-over.

### CODE REVIEW

File: blog/why-widgets.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+is computed when different font sizes are mixed:
+
+<iframe class="widget" src="widgets/lab3-baselines.html" height=204></iframe>
+
+Click the buttons to start the simulations and go through the steps
+one by one. You can see how baseline computation:
+
++ Finds the ascent and descent height of each individual word
++ Adds the maximum ascent to the _y_-cursor to compute the baseline
++ Places each word by moving it its ascent above the baseline
```
Comment (chrishtr) [inline blog/why-widgets.md:28]:
```suggestion
+ Places each word by moving its ascent above the baseline
```

File: blog/why-widgets.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
++ Computes the new _y_-cursor by adding the maximumn descent to the baseline
```
Comment (chrishtr) [inline blog/why-widgets.md:29]:
```suggestion
+ Computes the new _y_-cursor by adding the maximum descent to the baseline
```

File: blog/why-widgets.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+Second, the visual representation actually reinforces some key
+concepts. For example, layout has two phases, you _first_ lay it out
+horizontally, and only _later_ lay it out vertically. Putting two
+things on the screen, labeled "Phase 1" and "Phase 2", reinforces that.
+
+Third, the widget can actually reveal things that were hidden in the
+description. For example, the bullet points in the previous section
+didn't mention leading, but it clearly shows up in the widget.
+Moreover, the widget shows that the leading isn't evenly distributed
```
Comment (chrishtr) [inline blog/why-widgets.md:68]: What do you mean that it's not evenly distributed about the line?

File: blog/why-widgets.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+Second, the visual representation actually reinforces some key
+concepts. For example, layout has two phases, you _first_ lay it out
+horizontally, and only _later_ lay it out vertically. Putting two
+things on the screen, labeled "Phase 1" and "Phase 2", reinforces that.
+
+Third, the widget can actually reveal things that were hidden in the
+description. For example, the bullet points in the previous section
+didn't mention leading, but it clearly shows up in the widget. In
+fact, in a propert browser, the amount of leading above and below the
```
Comment (chrishtr) [inline blog/why-widgets.md:68]: Link to the chapter text for leading explaining the history and context

Comment (chrishtr) [reply blog/why-widgets.md:68]: Finish off this suggestion?

Comment (pavpanchekha) [reply blog/why-widgets.md:68]: I did link to it!

**Review by chrishtr — APPROVED** [review]

**Pull request merged**